tests/unit/s2n_connection_test.c

```diff
@@ -223,22 +242,37 @@ int main(int argc, char **argv)
         s2n_signature_algorithm inputs[] = { S2N_SIGNATURE_ANONYMOUS, S2N_SIGNATURE_RSA, 
                                              S2N_SIGNATURE_ECDSA, S2N_SIGNATURE_RSA_PSS_RSAE, 
                                              S2N_SIGNATURE_RSA_PSS_PSS };
-        /* If this fails, the S2N_SIGNATURE_SENTINEL value has changed, and a new enum has likely been added to s2n_handshake_signature_algorithm. */
         s2n_tls_signature_algorithm expected_output[] = { S2N_TLS_SIGNATURE_ANONYMOUS, S2N_TLS_SIGNATURE_RSA, 
                                                           S2N_TLS_SIGNATURE_ECDSA, S2N_TLS_SIGNATURE_RSA_PSS_RSAE, 
                                                           S2N_TLS_SIGNATURE_RSA_PSS_PSS };
         EXPECT_EQUAL(s2n_array_len(expected_output), s2n_array_len(inputs));
 
-        for (size_t i = 0; i < s2n_array_len(inputs); i++) {
-            conn->secure.client_cert_sig_scheme.sig_alg = inputs[i];
-            EXPECT_SUCCESS(s2n_connection_get_selected_client_cert_signature_algorithm(conn, &output));
-            conn->secure.client_cert_sig_scheme.sig_alg = 0;
-            EXPECT_EQUAL(expected_output[i], output);
-
-            conn->secure.conn_sig_scheme.sig_alg = inputs[i];
-            EXPECT_SUCCESS(s2n_connection_get_selected_signature_algorithm(conn, &output));
-            conn->secure.conn_sig_scheme.sig_alg = 0;
-            EXPECT_EQUAL(expected_output[i], output);
+        for (size_t i = 0; i < UINT16_MAX; i++) {
+            if (i < s2n_array_len(inputs) && s2n_is_valid_signature_algorithm(inputs[i])) {
+                conn->secure.client_cert_sig_scheme.sig_alg = inputs[i];
+                EXPECT_SUCCESS(s2n_connection_get_selected_client_cert_signature_algorithm(conn, &output));
+                EXPECT_EQUAL(expected_output[i], output);
+
+                conn->secure.conn_sig_scheme.sig_alg = inputs[i];
+                EXPECT_SUCCESS(s2n_connection_get_selected_signature_algorithm(conn, &output));
+                EXPECT_EQUAL(expected_output[i], output);
+            } else if (s2n_is_valid_signature_algorithm(i)) {
+                conn->secure.client_cert_sig_scheme.sig_alg = i;
```

Comment:
Sometimes you're testing i directly, sometimes you're testing inputs[i]. Does that guarantee you'll hit all possible values?

Maybe restructure your input like:
```
expected_output[] = {
  [ S2N_SIGNATURE_ANONYMOUS ] = S2N_TLS_SIGNATURE_ANONYMOUS,
  [ S2N_SIGNATURE_RSA ] = S2N_TLS_SIGNATURE_RSA,
  ...
};
```
Then you could consistently test i directly.

Reply:
It does guarantee that we will be hitting all possible values, I can clean it up to be more obvious/readable 👍